rpc/test/helpers.go

```diff
@@ -170,17 +179,70 @@ func NewTendermint(app abci.Application, opts *Options) *nm.Node {
 	if err != nil {
 		panic(err)
 	}
+
+	config.IPFS = cfg.DefaultIPFSConfig()
+	err = initIpfs(config, opts.loadIpfsPlugins, logger)
+	if err != nil {
+		panic(err)
+	}
 	node, err := nm.NewNode(config, pv, nodeKey, papp,
 		nm.DefaultGenesisDocProviderFunc(config),
 		nm.DefaultDBProvider,
 		nm.DefaultMetricsProvider(config.Instrumentation),
-		logger)
+		logger,
+		nm.IpfsPluginsWereLoaded(true),
+	)
 	if err != nil {
 		panic(err)
 	}
 	return node
 }
 
+func initIpfs(config *cfg.Config, loadPlugins bool, log log.Logger) error { // add counter part in ResetAllCmd
+	// init IPFS config with params from config.IPFS
+	// and store in config.IPFS.ConfigRootPath
+	repoRoot := config.IPFSRepoRoot()
+	if !fsrepo.IsInitialized(repoRoot) {
+		var conf *ipfscfg.Config
+
+		identity, err := ipfscfg.CreateIdentity(ioutil.Discard, []options.KeyGenerateOption{
+			options.Key.Type(options.Ed25519Key),
+		})
+		if err != nil {
+			return err
+		}
+		if err := tmos.EnsureDir(repoRoot, 0700); err != nil {
+			return err
+		}
+		if loadPlugins {
+			plugins, err := loader.NewPluginLoader(filepath.Join(repoRoot, "plugins"))
+			if err != nil {
+				return fmt.Errorf("error loading plugins: %s", err)
+			}
+			if err := plugins.Load(&nodes.LazyLedgerPlugin{}); err != nil {
+				return err
+			}
+			if err := plugins.Initialize(); err != nil {
+				return fmt.Errorf("error initializing plugins: %s", err)
+			}
+			if err := plugins.Inject(); err != nil {
+				return fmt.Errorf("error initializing plugins: %s", err)
+			}
+		}
+		conf, err = ipfscfg.InitWithIdentity(identity)
+		if err != nil {
+			return fmt.Errorf("intializing config failed, InitWithIdentity(): %w", err)
+		}
+
+		if err := fsrepo.Init(repoRoot, conf); err != nil {
+			return err
+		}
+	} else {
+		log.Info("ipfs repo already initialized", "repo-root", repoRoot)
+	}
+	return nil
+}
```

Comment:
The same thing that I mentioned above about de-nesting could be done here too, but it's up to you if you think it's worth it.